src/app.test.js

```diff
@@ -1,15 +1,42 @@
-// import { createApp } from "./app";
-// import userevent from '@testing-library/user-event'
-
-test("renders the todo app", () => {
-  //   // setup
-  //   const root = ...
-  //   const container = createApp(root)
-  //   // act
-  //   const input = queryByPlaceholderText('Please enter todo here')
-  //   userevent.type(input, 'new task')
-  //   const addButton = queryByText('Add')
-  //   userevent('click', addButton)
-  //   // assert
-  expect(true).tobee(true);
+import { screen, getByText, queryByText } from "@testing-library/dom";
+import userEvent from "@testing-library/user-event";
+import createApp from "./app";
+// import userEvent from "@testing-library/user-event";
+
+test("renders empty todo list when starts", () => {
+  // setup
+  const root = document.body;
+  createApp(root);
+  const todoList = screen.getByTestId("todo-list");
+  // act
+  // assert
+  expect(todoList).toBeEmptyDOMElement();
+});
+
+test("shows new item in todo list when added", () => {
+  // setup
+  const input = screen.getByPlaceholderText("Please enter todo here");
+  const addButton = screen.getByText("Add");
+
+  // act
+  userEvent.type(input, "Take wife for a walk!");
+  userEvent.click(addButton);
+
+  // assert
+  const todoList = screen.getByTestId("todo-list");
+  expect(queryByText(todoList, "Take wife for a walk!")).toBeInTheDocument();
+});
+
+test("removes an item from todo List when user clicks Delete button", () => {
+  // setup
```

Comment:
each test case must start from scratch and have their own specific setup to be independent of others

Reply:
an extra assertion might seem like a "duplicate" but could help to notice if something is not quite alright before the act or assertion phase, in this case:

```js
expect(queryByText(todoList, "Take wife for a walk!")).toBeInTheDocument();
```

which would fail without the appropriate state of the app being prepared.